packages/js/src/ui/components/InboxTabs/InboxTabs.tsx

```diff
@@ -0,0 +1,76 @@
+import { createSignal, onMount, Show } from 'solid-js';
+import { Tabs } from '../primitives';
+import { NotificationList } from '../Notification';
+import { MoreTabsDropdown } from './MoreTabsDropdown';
+import { InboxTab } from './InboxTab';
+
+type InboxTabsProps = {
+  tabs: Array<{ label: string; value: Array<string> }>;
+};
+
+export const InboxTabs = (props: InboxTabsProps) => {
+  const [activeTab, setActiveTab] = createSignal<string>((props.tabs[0] && props.tabs[0].label) ?? '');
+  const [tabsList, setTabsList] = createSignal<HTMLDivElement>();
+  const [visibleTabs, setVisibleTabs] = createSignal<InboxTabsProps['tabs']>([]);
+  const [dropdownTabs, setDropdownTabs] = createSignal<InboxTabsProps['tabs']>([]);
+
+  onMount(() => {
+    const tabsListEl = tabsList();
+    if (!tabsListEl) return;
+
+    const tabs = [...tabsListEl.querySelectorAll('[role="tab"]')];
+
+    const observer = new IntersectionObserver(
+      (entries) => {
+        let visibleTabIds = entries
+          .filter((entry) => entry.isIntersecting && entry.intersectionRatio === 1)
+          .map((entry) => entry.target.id);
+
+        if (tabs.length === visibleTabIds.length) {
+          setVisibleTabs(props.tabs.filter((tab) => visibleTabIds.includes(tab.label)));
+          observer.disconnect();
+          return;
+        }
+
+        visibleTabIds = visibleTabIds.slice(0, -1);
+        setVisibleTabs(props.tabs.filter((tab) => visibleTabIds.includes(tab.label)));
+        setDropdownTabs(props.tabs.filter((tab) => !visibleTabIds.includes(tab.label)));
+        observer.disconnect();
+      },
+      { root: tabsListEl }
+    );
+
+    for (const tabElement of tabs) {
+      observer.observe(tabElement);
+    }
+  });
+
+  return (
+    <Tabs class="nt-flex-1 nt-overflow-hidden" value={activeTab} onChange={setActiveTab}>
+      <Show
+        when={visibleTabs().length > 0}
+        fallback={
+          <Tabs.List ref={setTabsList}>
+            {props.tabs.map((tab) => (
+              <InboxTab label={tab.label} class="nt-invisible" />
+            ))}
+          </Tabs.List>
```

Comment:
These tabs are rendered hidden at the beginning (otherwise there would be a blink). Then we calculate which ones are fully visible and don't overflow in the parent container. Later we render the visible tabs.